syncthing.te

```diff
@@ -0,0 +1,45 @@
+policy_module(syncthing, 1.0.0)
+
+# define types
+type syncthing_t;
+type syncthing_exec_t;
+userdom_user_application_domain(syncthing_t, syncthing_exec_t)
+
+type syncthing_config_home_t;
+userdom_user_home_content(syncthing_config_home_t)
+
+# file permissions
+fs_getattr_xattr_fs(syncthing_t)
+
+userdom_manage_user_home_content(syncthing_t)
+
+# read randomness
+dev_read_rand(syncthing_t)
+dev_read_urand(syncthing_t)
+
+# networking
+userdom_basic_networking(syncthing_t)
+
+kernel_read_system_state(syncthing_t)
+kernel_read_net_sysctls(syncthing_t)
+
+sysnet_dns_name_resolve(syncthing_t)
+
+corenet_tcp_bind_generic_node(syncthing_t)
+corenet_udp_bind_generic_node(syncthing_t)
+
+corenet_tcp_bind_syncthing_port(syncthing_t)
+corenet_tcp_sendrecv_syncthing_port(syncthing_t)
+
+corenet_tcp_bind_syncthing_admin_port(syncthing_t)
+corenet_tcp_sendrecv_syncthing_admin_port(syncthing_t)
+
+corenet_udp_bind_syncthing_discovery_port(syncthing_t)
+corenet_udp_sendrecv_syncthing_discovery_port(syncthing_t)
+
+# file transitions
+syncthing_config_filetrans(syncthing_t)
+
+# roles
+attribute_role syncthing_roles;
+role syncthing_roles types syncthing_t syncthing_config_home_t;
```

Comment:
syncthing_roles need not be associated with syncthing_config_home_t. Since RBAC does not apply to objects.